packages/js-connect-with-portabl/src/index.ts

```diff
@@ -59,215 +98,380 @@ export class ConnectClient {
     );
   }
 
-  async _initiateTransaction(): Promise<void> {
-    const nonce = window.btoa(createRandomString());
+  private async _initiateTransaction(): Promise<void> {
+    const nonce: string = window.btoa(createRandomString());
 
-    const transactionData = await fetch(
+    const url: URL = new URL(
       `${this.options.connectDomain}/api/v1/agent/${this.options.accountId}/oauth2/transaction`,
-      {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify({
-          nonce,
-        }),
-      },
     );
 
-    const { authRequestUri, ...restTransaction } =
-      (await transactionData.json()) as CreateTransactionResponse;
-    const transformedAuthRequestUri = new URL(authRequestUri);
-    const clientId = transformedAuthRequestUri.searchParams.get('client_id');
+    const createTransactionResult: Response = await fetch(url.toString(), {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        nonce,
+      }),
+      // Note: handles error cases via redirect (302) in compliance with https://www.rfc-editor.org/rfc/rfc6749.html#section-4.2.2.1
+      redirect: 'follow',
+    });
+
+    const { authRequestUri, transactionId, state }: ICreateTransactionResponse =
+      await createTransactionResult.json();
+
+    const authRequestURL: URL = new URL(authRequestUri);
+
+    const clientId: string | null =
+      authRequestURL.searchParams.get('client_id');
 
-    if (typeof clientId !== 'string') {
-      throw new Error('Invalid clientId on authRequestUri');
+    if (
+      typeof transactionId !== 'string' ||
+      !transactionId.length ||
```

Comment:
Do we need to check for if these variable exists if we're already checking for typeof !== 'string'? We have the types for these as either a string or null then we can simplify to just checking for if its not a string